newIDE/app/src/Export/BrowserExporters/BrowserS3PreviewLauncher/index.js

```diff
@@ -41,6 +41,16 @@ export default class BrowserS3PreviewLauncher extends React.Component<
     };
   };
 
+  _markAsPreview = (fs: gdAbstractFileSystem, outputDir: string): void => {
```

Comment:
See my other comment, you should be able to do everything without adding yet another file manipulation.